src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -788,6 +818,27 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return emit_simd_ins_for_sig (cfg, klass, OP_CREATE_SCALAR, -1, arg0_type, fsig, args);
 	case SN_CreateScalarUnsafe:
 		return emit_simd_ins_for_sig (cfg, klass, OP_CREATE_SCALAR_UNSAFE, -1, arg0_type, fsig, args);
+	case SN_Equals:
+	case SN_EqualsAll:
+	case SN_EqualsAny: {
+		MonoType *arg_type = get_vector_t_elem_type (fsig->params [0]);
```

Comment:
We should start to add comprehensive vector element type checks for each case. (Both input args and the return.) I have a PR up to refactor the code of type checks for this function. (https://github.com/dotnet/runtime/pull/65486) You could merge this PR like this and fix it with your next PR.